modules/virt-about-aaq-operator.adoc

```diff
@@ -0,0 +1,77 @@
+
+
+.AAQ Operator controller and custom resources
+The AAQ Operator introduces two new API objects defined as custom resource definitions (CRDs) for managing alternative quota implementations across multiple namespaces:
+
+* `ApplicationAwareResourceQuota`: Sets aggregate quota restrictions enforced per namespace. The `ApplicationAwareResourceQuota` API is compatible with the native `ResourceQuota` object and shares the same specification and status definitions.
++
+.Example manifest
+[source,yaml]
+----
+apiVersion: aaq.kubevirt.io/v1alpha1
+kind: ApplicationAwareResourceQuota
+metadata:
+  name: example-resource-quota
+spec:
+  hard:
+    requests.memory: 1Gi 
+    limits.memory: 1Gi
+    requests.cpu/vmi: "1" # <1>
+    requests.memory/vmi: 1Gi # <2>
+# ...
+----
+<1> For all pods in a non-terminal state that have VM workloads, the minimum amount of CPU that is reserved for the VM container.
+<2> For all pods in a non-terminal state that have VM workloads, the minimum amount of RAM that is reserved for the VM container.
+
+* `ApplicationAwareClusterResourceQuota`: Mirrors the `ApplicationAwareResourceQuota` object at a cluster scope. It is compatible with the native `ClusterResourceQuota` API object and shares the same specification and status definitions. AAQ cluster quotas are not namespaced. When creating an AAQ cluster quota, you can select multiple namespaces based on annotation selection, label selection, or both by editing the `spec.selector.labels` or `spec.selector.annotations` fields.
++
+.Example manifest
+[source,yaml]
+----
+apiVersion: aaq.kubevirt.io/v1alpha1
+kind: ApplicationAwareClusterResourceQuota # <1>
+metadata:
+  name: example-resource-quota
+spec:
+  quota:
+    hard:
+    requests.memory: 1Gi
+    limits.memory: 1Gi
+    requests.cpu/vmi: "1"
+    requests.memory/vmi: 1Gi
+  selector:
+    annotations: null
+    labels:
+      matchLabels:
+        kubernetes.io/metadata.name: default
+# ...
+----
+<1> You can only create an `ApplicationAwareClusterResourceQuota` object if the `spec.allowApplicationAwareClusterResourceQuota` field in the `HyperConverged` CR is set to `true`.
++
+[NOTE]
+====
+If both `spec.selector.labels` and`spec.selector.annotations` fields are set, only namespaces that match both are selected.
+====
+
+The AAQ controller uses a scheduling gate mechanism to evaluate whether there is enough of a resource available to run a VM pod. If so, the scheduling gate is removed from the pod and it is considered ready for scheduling. The quota usage status is updated to indicate how much of the quota is used. If the CPU and memory requests and limits for the VM pod exceed the enforced quota usage limit, pod creation fails.
```

Comment:
> The AAQ controller uses a scheduling gate mechanism to evaluate whether there is enough of a resource available to run a VM pod

To run a vm workload or another workload

 > If the CPU and memory requests and limits for the VM pod exceed the enforced quota usage limit, pod creation fails.

 If the CPU and memory requests and limits for the workload exceed the enforced quota usage limit, pod won't be scheduled until there is enough quota.